File: hanjo/개인용/programmers/완전탐색/L2_카펫/solution.py
import math
def solution(brown, yellow):
    grid_num = brown + yellow
    for x in range(math.ceil(math.sqrt(grid_num)), grid_num-1, 1):
        y, remain = divmod(grid_num, x)
        if remain==0 and (x+(y-2))*2 == brown:
            return [x,y]
            

brown = 10
yellow = 2
print(solution(brown, yellow))

brown = 8
yellow = 1
print(solution(brown, yellow))

brown = 24
yellow = 24
print(solution(brown, yellow))

File: hanjo/개인용/programmers/완전탐색/L2_카펫/Solution.java
package 완전탐색.L2_카펫;

import java.util.*;

public class Solution {
	public static int[] solution(int brown, int yellow) {
		int[] answer = new int[2];
		
		int gridNum = brown + yellow;
		int firstX = (int)Math.ceil(Math.sqrt(gridNum));
		
		for(int x=firstX; x<=gridNum; x++) {
			if(gridNum%x==0) {
				int y = gridNum/x;
				if((x+(y-2))*2 == brown) {
					answer[0] = x;
					answer[1] = y;
					break;
				}
			}
		}
		return answer;
    }

	public static void main(String[] args) {
		int brown = 10;
		int yellow = 2;
		System.out.println(Arrays.toString(solution(brown, yellow)));
	}
}
